flycheck.el

```diff
@@ -8711,6 +8715,25 @@ rustc command.  See URL `https://www.rust-lang.org'."
                (and (flycheck-buffer-saved-p)
                     (locate-dominating-file (buffer-file-name) "Cargo.toml"))))
 
+(flycheck-define-checker rust-cargo-clippy
+  "A Rust syntax checker and linter using clippy.
+
+This syntax checker needs the cargo clippy subcommand. See URL `https://github.com/Manishearth/rust-clippy'"
+  :command ("cargo" "clippy"
+            (eval flycheck-cargo-clippy-args)
+            "--"  "--error-format=json"
+            (option-list "-L" flycheck-rust-library-path concat)
+            (eval flycheck-rust-args))
```

Comment:
So we'll have three argument variables.

- `flycheck-rust-args` that are for the compiler (`rustc`)
- `flycheck-cargo-rustc-args` are arguments to the `cargo rustc` *command*, not to the compiler.
- `flycheck-cargo-clippy-args` are arguments to the `cargo clippy` command, and `cargo clippy` accepts a superset of the options of `cargo rustc`.

It's getting a bit confusing, but at least the naming is consistent ;)